### DISCUSSION TITLE
How do I fix my existing python Bot apps after this annoying upgrade

### DISCUSSION THREAD

**Discussion by TheSoldiersDream** [openai/openai-python#782] [Q&A]
# Read OpenAI API key from environment variable
openai.api_key = what has changed now that I need to do in my code to get my advanced bots to work correctly under API keys?
Can anyone please give me some step by step instructions as I dont follow the existing merge document or upgrade what ever you call it?

**Comment by lostmygithubaccount** [comment]
this is a very unclear question, but you probably want to pin the version of the Python client you are using to avoid breaking changes

**Reply by GCODIN** [reply]
> this is a very unclear question, but you probably want to pin the version of the Python client you are using to avoid breaking changes

>new to GitHub and trying to figure out what to use <
